component/src/test/java/io/siddhi/extension/io/file/FileHandlingTestCase.java

```diff
@@ -0,0 +1,384 @@
+package io.siddhi.extension.io.file;
+
+import io.siddhi.core.SiddhiAppRuntime;
+import io.siddhi.core.SiddhiManager;
+import io.siddhi.core.event.Event;
+import io.siddhi.core.exception.SiddhiAppCreationException;
+import io.siddhi.core.stream.output.StreamCallback;
+import io.siddhi.core.util.EventPrinter;
+import io.siddhi.core.util.SiddhiTestHelper;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.TestException;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class FileHandlingTestCase {
+    private static final Logger log = Logger.getLogger(FileSourceTextFullModeTestCase.class);
+    private AtomicInteger count = new AtomicInteger();
+
+    private File newRoot;
+
+    @BeforeClass
+    public void init() {
+        ClassLoader classLoader = FileHandlingTestCase.class.getClassLoader();
+        String rootPath = Objects.requireNonNull(classLoader.getResource("files")).getFile();
+        String dirUri = rootPath + "/new";
+        newRoot = new File(dirUri);
+    }
+
+    @BeforeMethod
+    public void doBeforeMethod() throws TestException {
+        count.set(0);
+        try {
+            FileUtils.forceMkdir(newRoot);
+        } catch (IOException e) {
+            throw new TestException("Failed to make directory " + newRoot, e);
+        }
+    }
+
+    @AfterMethod
+    public void doAfterMethod() {
+        try {
+            FileUtils.forceDelete(newRoot);
+        } catch (IOException e) {
+            throw new TestException("Failed to delete files in the " + newRoot, e);
+        }
+        count.set(0);
+    }
+
+    /**
+     * Test cases for 'type = 'file_handler'
+     * */
+    @Test
+    public void siddhiIOFileTest1() throws InterruptedException {
+        log.info("Siddhi IO File Testcase 1 - Directory");
+        String app = "" +
+                "@App:name('TestSiddhiApp') @source(type='fileeventlistener', " +
+                "uri='file:" + newRoot + "', @map(type='json'))\n" +
+                "define stream CreateFileStream(filepath string, length long, last_modified string, status string);\n" +
+                "define stream ResultStream(filepath string, length long, last_modified string, status string);\n" +
+                "from CreateFileStream\n" +
+                "select *\n" +
+                "insert into ResultStream;";
+
```

Comment:
Lets remove unnecessary spaces. Check all other places as well.

Reply:
fixed in 3f9a2a0